lustre-collector/src/main.rs

```diff
@@ -113,36 +112,17 @@ fn run() -> Result<(), LustreCollectorError> {
         .get_one::<Format>("format")
         .expect("Required argument `format` missing");
 
-    let handle = thread::spawn(move || -> Result<Vec<Record>, LustreCollectorError> {
-        let lctl_output = get_lctl_output()?;
+    let lctl_output = get_lctl_output()?;
+    let mut lctl_record = parse_lctl_output(&lctl_output)?;
 
-        let lctl_record = parse_lctl_output(&lctl_output)?;
+    let lctl_mgs_output = get_lctl_mgs_fs_output()?;
+    let mut lctl_mgs_record = parse_mgs_fs_output(&lctl_mgs_output)?;
 
-        Ok(lctl_record)
-    });
+    let lnetctl_stats_output = get_lnetctl_stats_output()?;
+    let mut lnetctl_stats_record = parse_lnetctl_stats(str::from_utf8(&lnetctl_stats_output)?)?;
 
-    let mgs_fs_handle = thread::spawn(move || -> Result<Vec<Record>, LustreCollectorError> {
-        let lctl_output = get_lctl_mgs_fs_output()?;
-        let lctl_record = parse_mgs_fs_output(&lctl_output)?;
-
-        Ok(lctl_record)
-    });
-
-    let lnetctl_stats_handle =
-        thread::spawn(move || -> Result<Vec<Record>, LustreCollectorError> {
-            let lnetctl_stats_output = get_lnetctl_stats_output()?;
-            let lnetctl_stats_record = parse_lnetctl_stats(str::from_utf8(&lnetctl_stats_output)?)?;
-
-            Ok(lnetctl_stats_record)
-        });
-
-    let recovery_status_handle =
-        thread::spawn(move || -> Result<Vec<Record>, LustreCollectorError> {
-            let recovery_status_output = get_recovery_status_output()?;
-            let recovery_statuses = parse_recovery_status_output(&recovery_status_output)?;
-
-            Ok(recovery_statuses)
-        });
```

Comment:
Removed `thread::spawn` as the results is now referencing a variable local to the thread.
I tried to use `Arc` without success. 

@jgrund any idea?